node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -459,13 +462,314 @@ fn validate_lnc_two_equivocators() -> Result<(), AddUnitError<TestContext>> {
+        panorama: panorama!(a0, b0, c0, F, e0, f0, g0, N),
+        creator: HANNA,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 52.into(),
+        round_exp: 4u8,
+        endorsed: vec![a0, g0].into_iter().collect(), // Endorsements included in the panorama.
+    };
+    assert_eq!(state.validate_lnc(&h0), Some(DAN));
+    // Even though both a0 and g0 cite DAN non-naively, for h0 to be valid
+    // we need to endorse either c0 or e0.
+    let mut pre_endorse_state = state.clone();
+    // Endorse b0 first.
+    endorse!(state, rng, c0; ALICE, BOB, CAROL, ERIC, FRANK);
+    h0.endorsed.insert(c0);
+    assert_eq!(state.validate_lnc(&h0), None);
+    // Should also pass if e0 is endorsed.
+    endorse!(pre_endorse_state, rng, e0; ALICE, BOB, CAROL, ERIC, FRANK);
+    h0.endorsed.insert(e0);
+    assert_eq!(pre_endorse_state.validate_lnc(&h0), None);
+    Ok(())
+}
+
+#[test]
+#[ignore = "stubbed validate_lnc"]
+fn validate_lnc_four_forks() -> Result<(), AddUnitError<TestContext>> {
+    // Diagram of the DAG can be found under
+    // /resources/test/dags/validate_lnc_four_forks.png
+    let weights = &[
+        Weight(3),
+        Weight(4),
+        Weight(5),
+        Weight(5),
+        Weight(5),
+        Weight(5),
+        Weight(5),
+        Weight(5),
+    ];
+    let mut state = State::new_test(weights, 0);
+    let mut rng = crate::new_rng();
+    let e0 = add_unit!(state, rng, ERIC, 0xEA; N, N, N, N, N, N, N, N)?;
+    let e0_prime = add_unit!(state, rng, ERIC, 0xEB; N, N, N, N, N, N, N, N)?;
+    let e0_bis = add_unit!(state, rng, ERIC, 0xEC; N, N, N, N, N, N, N, N)?;
+    let e0_cis = add_unit!(state, rng, ERIC, 0xED; N, N, N, N, N, N, N, N)?;
+    let a0 = add_unit!(state, rng, ALICE, None; N, N, N, N, e0, N, N, N)?;
+    let b0 = add_unit!(state, rng, BOB, None; N, N, N, N, e0_prime, N, N, N)?;
+    let g0 = add_unit!(state, rng, GINA, None; N, N, N, N, e0_bis, N, N, N)?;
+    let h0 = add_unit!(state, rng, HANNA, None; N, N, N, N, e0_cis, N, N, N)?;
+    endorse!(state, rng, a0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
+    let c0 = add_unit!(state, rng, CAROL, None; a0, b0, N, N, F, N, N, N; a0)?;
+    endorse!(state, rng, g0; ALICE, BOB, CAROL, DAN, GINA, HANNA);
+    let f0 = add_unit!(state, rng, FRANK, None; N, N, N, N, F, N, g0, h0; g0)?;
+    let d0 = add_unit!(state, rng, DAN, None; N, N, N, N, F, f0, g0, h0)?;
+    let mut d1 = WireUnit {
+        panorama: panorama!(a0, b0, c0, d0, F, f0, g0, h0),
+        creator: DAN,
+        instance_id: 1u64,
+        value: None,
+        seq_number: 0,
+        timestamp: 52.into(),
+        round_exp: 4u8,
+        endorsed: vec![a0, g0].into_iter().collect(), // Endorsements included in the panorama.
+    };
+    assert_eq!(state.validate_lnc(&d1), Some(ERIC));
+    let mut pre_endorse_state = state.clone();
+    // If we endorse h0, then d1 should be OK since there won't be two pair of equivocating units
```

Comment:
```suggestion
    // If we endorse h0, then d1 should be OK since there won't be two equivocating units
```

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/535/commits/de7e6599a16c1ec099c31e093be5113caeca8647.